tests/Unit/Validators/ApplicationTimelineValidatorTest.php

```diff
@@ -0,0 +1,253 @@
+    }
+
+    public function testFitComplete(): void
+    {
+        $applicant = factory(Applicant::class)->create();
+        $validator = new ApplicationTimelineValidator();
+
+        // Factory should create a-complete application.
+        $completeApplication = factory(JobApplication::class)->states(['submitted', 'version2'])->create([
+            'applicant_id' => $applicant->id
+        ]);
+        $applicant->job_applications()->save($completeApplication);
+        $this->assertTrue($validator->fitComplete($completeApplication));
+
+        // Removing Answers should invalidate fit step.
+        $incompleteApplication = factory(JobApplication::class)->states(['submitted', 'version2'])->create([
+            'applicant_id' => $applicant->id
+        ]);
+        $incompleteApplication->job_application_answers()->delete();
+        $this->assertFalse($validator->fitComplete($incompleteApplication));
+
+        // Having one answer empty should invalidate fit step.
+        $wrongApp1 = factory(JobApplication::class)->create([
+            'applicant_id' => $applicant->id
+        ]);
+        $answer = $wrongApp1->job_application_answers->first();
+        $answer->answer = '';
+        $answer->save();
+        $this->assertFalse($validator->fitComplete($wrongApp1));
+
+        // Having one answer over word count should invalidate fit step.
+        $wrongApp1 = factory(JobApplication::class)->create([
+            'applicant_id' => $applicant->id
+        ]);
+        $answer = $wrongApp1->job_application_answers->first();
+        $answer->answer = $this->faker->words(255, true);
+        $answer->save();
+        $this->assertFalse($validator->fitComplete($wrongApp1));
+
+        // Having an extra answer, related to the wrong job, should invalidate.
+        $wrongApp2 = factory(JobApplication::class)->create([
+            'applicant_id' => $applicant->id
+        ]);
+        $job2 = factory(JobPoster::class)->create();
+        $wrongApp2->job_application_answers()->save(factory(JobApplicationAnswer::class)->create([
+            'job_application_id' => $wrongApp2->id,
+            'job_poster_question_id' => $job2->job_poster_questions->first()->id
+        ]));
+        $this->assertFalse($validator->fitComplete($wrongApp2));
+    }
+
+    public function testExperienceComplete(): void
+    {
+
+        $validator = new ApplicationTimelineValidator();
+        // Draft Application
+        $draftApplication = factory(JobApplication::class)->states(['draft', 'version2'])->create();
+        // Every essential criterion has an assigned experience above, this should
+        // validate.
+        $this->assertTrue($validator->experienceComplete($draftApplication));
+        // Removing one or more experiences from a required criterion should fail
+        // validation.
+        $essentialCriteriaType = CriteriaType::where('name', 'essential')->first()->id;
+        $requiredSkill = $draftApplication->job_poster->criteria
```

Comment:
When looking for a required skill (here and on other tests in the file) make sure to check that its a 'hard' skill as well. As is, it will occasionally fail.